cypress/integration/01-slate-zotero-block.js

```diff
@@ -0,0 +1,540 @@
+    cy.get('.slate-editor.selected [contenteditable=true]').not(
+      'span[id^="cite_ref"]',
+    );
+  });
+
+  it('Add Zotero citations', () => {
+    // intercept requests to simulate response and not use real credentials
+    cy.intercept('GET', '**/@zotero', {
+      statusCode: 200,
+      body: {
+        default: 'NH578GBA',
+        password: 'test',
+        server: 'https://api.zotero.org/users/6732',
+        style: 'https://www.eea.europa.eu/zotero/eea.csl',
+      },
+    });
+    // the top two collections
+    cy.intercept(
+      'GET',
+      'https://api.zotero.org/users/6732/collections/top/?start=0&limit=10&sort=title',
+      {
+        statusCode: 200,
+        headers: {
+          'Total-Results': '14',
+        },
+        body: [
+          {
+            key: '24C33',
+            version: 145,
+            data: {
+              key: '24C33',
+              version: 145,
+              name: '2009',
+              parentCollection: false,
+              relations: {},
+            },
+            library: {
+              type: 'user',
+              id: 6732,
+              name: 'rexalexmede',
+              links: {
+                alternate: {
+                  href: 'https://www.zotero.org/rexalexmede',
+                  type: 'text/html',
+                },
+              },
+              self: {
+                href: 'https://api.zotero.org/users/6732/collections/24C33',
+                type: 'application/json',
+              },
+            },
+            meta: { numCollections: 0, numItems: 2 },
+          },
+          {
+            key: 'DH68XG',
+            version: 144,
+            library: {
+              type: 'user',
+              id: 6732,
+              name: 'rexalexmede',
+              links: {
+                alternate: {
+                  href: 'https://www.zotero.org/rexalexmede',
+                  type: 'text/html',
+                },
+              },
+            },
+            links: {
+              self: {
+                href: 'https://api.zotero.org/users/6732/collections/DH68XG',
+                type: 'application/json',
+              },
+              alternate: {
+                href: 'https://www.zotero.org/rexalexmede/collections/DH68XG',
+                type: 'text/html',
+              },
+            },
+            meta: {
+              numCollections: 0,
+              numItems: 1,
+            },
+            data: {
+              key: 'DH68XG',
+              version: 144,
+              name: '2010',
+              parentCollection: false,
+              relations: {},
+            },
+          },
+        ],
+      },
+    );
+    // the two items in the first collection
+    cy.intercept(
+      'GET',
+      'https://api.zotero.org/users/6732/collections/24C33/items/?start=0&limit=10&sort=title',
+      {
+        statusCode: 200,
+        headers: {
+          'Total-Results': '14',
+        },
+        body: [
```

Comment:
You can also move this inside `fixtures/` to make it look more concise.

Reply:
fixed